ui-tests/src/test/java/io/syndesis/qe/pages/connections/ConnectionsListComponent.java

```diff
@@ -0,0 +1,95 @@
+package io.syndesis.qe.pages.connections;
+
+import static com.codeborne.selenide.CollectionCondition.sizeGreaterThanOrEqual;
+import static com.codeborne.selenide.Condition.visible;
+import static com.codeborne.selenide.Selenide.$;
+
+import org.openqa.selenium.By;
+
+import com.codeborne.selenide.ElementsCollection;
+import com.codeborne.selenide.SelenideElement;
+
+import java.util.List;
+
+import io.syndesis.qe.pages.SyndesisPageObject;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Created by sveres on 11/10/17.
+ */
+@Slf4j
+public class ConnectionsListComponent extends SyndesisPageObject {
+
+	private static final class Button {
+		public static final By DELETE = By.linkText("Delete");
+		public static final By OK = By.linkText("Ok");
+	}
+
+	private static final class Element {
+		public static final By ROOT = By.cssSelector("syndesis-connections-list");
+	}
+
+	@Override
+	public SelenideElement getRootElement() {
+		return $(Element.ROOT).shouldBe(visible);
+	}
+
+	@Override
+	public boolean validate() {
+		return false;
+	}
+
+	public Integer countConnections() {
+		ElementsCollection allConnections =
+				$(Element.ROOT).findAll(By.cssSelector("h2.card-pf-title.text-center")).shouldBe(sizeGreaterThanOrEqual(0));
+		log.info(String.format("found %d connections", allConnections.size()));
+		return allConnections.size();
+	}
+
+	public SelenideElement getConnectionByTitle(String title) {
+		log.info(String.format("searching for connection %s", title));
+		return $(Element.ROOT).find(By.cssSelector(String.format("div.connection[title = \"%s\"]", title)));
+	}
+
+	public void goToConnection(String title) {
+		log.info(String.format("searching for connection %s", title));
+		this.getConnectionByTitle(title).shouldBe(visible).click();
+	}
+
+	public void deleteConnection(String title) {
+		log.info(String.format("searching delete link for connection %s", title));
+		this.getConnectionByTitle(title).find(By.id("dropdownKebabRight9")).shouldBe(visible).click();
+		$(Element.ROOT).find(Button.DELETE).shouldBe(visible).click();
+		$(Element.ROOT).find(Button.OK).shouldBe(visible).click();
+	}
+
+	public ElementsCollection getAllKebabElements(Boolean isOpen) {
+		String open = isOpen ? ".open" : "";
+		return $(Element.ROOT).findAll(By.cssSelector(String.format("div.dropdown.dropdown-kebab-pf.pull-right%s}", open))).shouldBe(sizeGreaterThanOrEqual(0));
+	}
+
+	public ElementsCollection getAllKebabButtons() {
+		return $(Element.ROOT).findAll(By.cssSelector("button.btn.btn-link.dropdown-toggle")).shouldBe(sizeGreaterThanOrEqual(0));
```

Comment:
It would be nice to have all these locator objects statically defined similar way as root locator.

Reply:
actually I had it in 'statically defined' but during todays standup meeting we agreed to keep it as function :-)

Reply:
I think you get it wrong, locators should be there.

Reply:
done!